app.py

```diff
@@ -226,43 +231,97 @@ def disable_form():
     st.session_state.in_progress = True
 
 
+def initialize_session():
+    if "authenticated" not in st.session_state:
+        st.session_state["authenticated"] = False
+    if "login_time" not in st.session_state:
+        st.session_state["login_time"] = None
+
+
+def is_authenticated():
+    if not st.session_state["authenticated"]:
+        return False
+    if st.session_state["login_time"]:
+        if datetime.now() - st.session_state["login_time"] > timedelta(days=1):
```

Comment:
If you want to set it, it would be better to be able to set it in hours or minutes and change it from an environment variable.

Reply:
Yeah, agreed. It was a very generic expiring time. In case you need it, it can be easily changed. I will set 60 minutes by default, what do you think?

Reply:
@ryo-ma , heads up, the module that you preferred uses `days` for expiring time

>- cookie_expiry_days: float, default 30.0
>
>      Specifies the number of days before the re-authentication cookie automatically expires on the client's browser.

Realistically, this isn't a crucial implementation for the app's usage context, which is primarily for personal use. It's good to have, but it shouldn't be a blocker. So, I'll leave it for you. I'll just implement the suggested module as-is.